spec/support/api/helpers.rb

```diff
@@ -22,7 +22,14 @@ def init_api_spec_env
         def api_basic_authorize(*identifiers, user: @user.userid, password: @user.password)
           if identifiers.present?
             identifiers.flatten.collect do |identifier|
-              @role.miq_product_features << MiqProductFeature.find_or_create_by(:identifier => identifier) if identifier
+              if identifier
+                # TODO:  We create product features on demand for various tests instead of seeding the correct features.
+                # We shouldn't hardcode a feature_type here but since it's a UI concern and it's now required, it shouldn't affect the API.
+                # See also: https://github.com/ManageIQ/manageiq/pull/21207
+                feature = MiqProductFeature.find_by(:identifier => identifier)
+                feature ||= MiqProductFeature.create(:identifier => identifier, :feature_type => "view")
+                @role.miq_product_features << feature
+              end
```

Comment:
Minor, but we can do this with Rails directly

```suggestion
              if identifier
                # TODO:  We create product features on demand for various tests instead of seeding the correct features.
                # We shouldn't hardcode a feature_type here but since it's a UI concern and it's now required, it shouldn't affect the API.
                # See also: https://github.com/ManageIQ/manageiq/pull/21207
                @role.miq_product_features = MiqProductFeature.create_with(:feature_type => "view").find_or_create_by(:identifier => identifier)
              end
```

Reply:
Even so, I'm going to merge as is